### PR TITLE
Feature/process pfflow

### PR DESCRIPTION
**Pull request by jrouzierinverse**
# Description
Add logic for processing pfflows in pfcron.

# Delete branch after merge
YES

# Checklist
(REQUIRED) - [yes, no or n/a]
- [ ] Document the feature
- [ ] Add OpenAPI specification
- [ ] Add unit tests
- [ ] Add acceptance tests (TestLink)

### CODE REVIEW

File: html/pfappserver/lib/pfappserver/Form/Config/Pfcron/pfflow.pm [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+use pfappserver::Form::Config::Pfcron qw(default_field_method batch_help_text timeout_help_text window_help_text);
+
+extends 'pfappserver::Form::Config::Pfcron';
+with 'pfappserver::Base::Form::Role::Help';
+
+has_field 'kafka_brokers' => (
+    type => 'Text',
+    default_method => \&default_field_method,
+    tags => { after_element => \&help,
+             help => \&batch_help_text },
```
Comment (fdurand) [inline html/pfappserver/lib/pfappserver/Form/Config/Pfcron/pfflow.pm]: Probably useless:
```
  sub timeout_help_text { "Maximum amount of time this task can run." }
```

File: html/pfappserver/lib/pfappserver/Form/Config/Pfcron/pfflow.pm [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+);
+
+has_field 'read_topic' => (
+    type => 'Text',
+    default_method => \&default_field_method,
+    tags => { after_element => \&help,
+             help => \&timeout_help_text },
```
Comment (fdurand) [inline html/pfappserver/lib/pfappserver/Form/Config/Pfcron/pfflow.pm]: Same

File: html/pfappserver/lib/pfappserver/Form/Config/Pfcron/pfflow.pm [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+);
+
+has_field 'send_topic' => (
+    type => 'Text',
+    default_method => \&default_field_method,
+    tags => { after_element => \&help,
+             help => \&timeout_help_text },
```
Comment (fdurand) [inline html/pfappserver/lib/pfappserver/Form/Config/Pfcron/pfflow.pm]: Same

File: html/pfappserver/lib/pfappserver/Form/Config/Pfcron/pfflow.pm [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+);
+
+has_field 'group_id' => (
+    type => 'Text',
+    default_method => \&default_field_method,
+    tags => { after_element => \&help,
+             help => \&timeout_help_text },
```
Comment (fdurand) [inline html/pfappserver/lib/pfappserver/Form/Config/Pfcron/pfflow.pm]: Same

File: html/pfappserver/lib/pfappserver/Form/Config/Pfcron/pfflow.pm [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+);
+
+has_field 'submit_batch' => (
+    type => 'PosInteger',
+    default_method => \&default_field_method,
+    tags => { after_element => \&help,
+             help => \&timeout_help_text },
```
Comment (fdurand) [inline html/pfappserver/lib/pfappserver/Form/Config/Pfcron/pfflow.pm]: Same

File: lib/pf/iptables.pm [hunk 6]
```diff
@@ -237,15 +238,21 @@ generate_kafka_rules
...
-        $$rule .= "-A input-management-if --protocol tcp --match tcp -s $ip --dport 9092 --jump ACCEPT\n";
-        $$rule .= "-A input-management-if --protocol tcp --match tcp -s $ip --dport 9093 --jump ACCEPT\n";
-    }
+    return if !exists $ConfigKafka{hostname()};
+
+    $$rule .= "-A input-management-if --protocol tcp --match tcp --dport 9092 --jump ACCEPT\n";
+    $$rule .= "-A input-management-if --protocol tcp --match tcp --dport 9093 --jump ACCEPT\n";
+    $$rule .= "-A input-management-if --protocol tcp --match tcp --dport 29092 --jump ACCEPT\n";
+
+#    for my $client (@{$ConfigKafka{iptables}{clients}}) {
```
Comment (fdurand) [inline lib/pf/iptables.pm:247]: Remove

File: go/cron/kafka_submitter.go [hunk 7]
```diff
@@ -0,0 +1,184 @@
...
+
+func (s *KafkaSubmiter) shutdown() {
+	s.writer.Close()
+}
+
+func (s *KafkaSubmiter) send(events []*NetworkEvent) {
+	var filteredEvents []*NetworkEvent
+	if s.filterEvents {
+		db, err := getDb()
+		if err != nil {
```
Comment (fdurand) [inline go/cron/kafka_submitter.go:107]: Do we want to log the error ?

File: go/cron/kafka_submitter.go [hunk 8]
```diff
@@ -0,0 +1,184 @@
...
+
+	} else {
+		filteredEvents = events
+	}
+
+	messages := make([]kafka.Message, 0, len(filteredEvents))
+	for i := 0; i < len(filteredEvents); i++ {
+		data, err := json.Marshal(events[i])
+		if err != nil {
+			//TODO log error
```
Comment (fdurand) [inline go/cron/kafka_submitter.go:131]: TODO Log error

File: go/cron/pfflow.go [hunk 9]
```diff
@@ -0,0 +1,158 @@
+package maint
+
+import (
+	"net/netip"
+	"time"
+)
+
+//easyjson:json
```
Comment (fdurand) [inline go/cron/pfflow.go:8]: Remove

File: go/cron/pfflow.go [hunk 10]
```diff
@@ -0,0 +1,158 @@
...
+	DomainID       uint32     `json:"domain_id"`
+	SubAgentID     uint32     `json:"sub_agent_id"`
+	SysUptime      uint32     `json:"sys_uptime"`
+	FlowType       uint16     `json:"flow_type"`
+	SampleInterval uint16     `json:"sample_interval"`
+	EngineType     uint8      `json:"engine_type"`
+	EngineId       uint8      `json:"engine_id"`
+}
+
+//easyjson:json
```
Comment (fdurand) [inline go/cron/pfflow.go:22]: Remove

File: go/cron/pfflow.go [hunk 11]
```diff
@@ -0,0 +1,158 @@
...
+type PfFlows struct {
+	Header PfFlowHeader `json:"header"`
+	Flows  *[]PfFlow    `json:"flows"`
+}
+
+//easyjson:json
```
Comment (fdurand) [inline go/cron/pfflow.go:28]: Remove

File: rpm/packetfence.spec [hunk 12]
```diff
@@ -1038,6 +1038,8 @@ fi
 %dir                    /usr/local/pf/conf/I18N
 %dir                    /usr/local/pf/conf/I18N/api
                         /usr/local/pf/conf/I18N/api/*
+%dir                    /usr/local/pf/conf/kafka
+%config                 /usr/local/pf/conf/kafka/kafka_server_jaas.conf.tt
 %dir                    /usr/local/pf/conf/locale
```
Comment (fdurand) [inline rpm/packetfence.spec:1044]: You should add kafka.conf too as a config file (when we do an upgrade it will override it)

Comment (fdurand) [reply rpm/packetfence.spec:1044]: Also in debian/packetfence.conffiles

**Review by fdurand — COMMENTED** [review]
Do we have a way to manage the Kafka.conf file from the admin ui ?

**Comment by jrouzierinverse** [comment]
> Do we have a way to manage the Kafka.conf file from the admin ui ?

Not at the moment

File: go/cron/pfflow.go (hunk 10 above)
Comment (jrouzierinverse) [reply go/cron/pfflow.go:22]: This is needed for code generation with easyjson

File: go/cron/pfflow.go (hunk 9 above)
Comment (jrouzierinverse) [reply go/cron/pfflow.go:8]: This is needed for code generation with easyjson

File: go/cron/pfflow.go (hunk 11 above)
Comment (jrouzierinverse) [reply go/cron/pfflow.go:28]: This is needed for code generation with easyjson

**Pull request merged**